### PR TITLE
br: add encryption config to streaming backup

### PR DESCRIPTION
**Pull request by Tristan1900**
1. In stream back up task, add an optional plaintext data key, the master key based in set up in stream backup config when cluster started.
2. Add a new message `FileEncryptionInfo` to store encrypted data key information, add it to `DataFileInfo` that's uploaded to S3 as the backup file metadata.
3. expand `MasterKeyKms` with optional GCP and Azure KMS information so can used to construct master key backend
4. Add the `MasterKey` to `ApplyRequest` so during log restore master key information can be passed down to TiKV.
5. Changed the `check.sh` to accommodate version format `major.minor` besides `major.minor.patch`, add checks for protobuf 27 major release version after major version 3

### CODE REVIEW

**Comment by Tristan1900** [comment]
please don't merge yet, pending on design approval

**Comment by Tristan1900** [comment]
pending on design review, please do not merge

**Comment by Tristan1900** [comment]
WIP, not ready to merge

File: proto/brpb.proto
Comment (YuJuncen) [inline proto/brpb.proto:676]: This isn't a logical checksum over raw kvs, the checksum was executed after compression if i recall right.

File: scripts/check.sh
Comment (YuJuncen) [inline scripts/check.sh:12]: Thank you! Now we no more need to download an old `protoc` for passing the check...

File: proto/brpb.proto
Comment (Tristan1900) [reply proto/brpb.proto:676]: discussed offline, this is pre-compression checksum of the file but not exactly logical too. It's the checksum of the plaintext file.

**Comment by Tristan1900** [comment]
hey folks, @BornChanger @overvenus, could you take a look at this change when you get a chance? Thanks!

File: proto/import_sstpb.proto
Comment (overvenus) [inline proto/import_sstpb.proto:400]: The largest tag number in this message is 13. Why 19?

File: proto/encryptionpb.proto
Comment (overvenus) [inline proto/encryptionpb.proto:156]: Empty message, how to pass data keys?

File: proto/import_sstpb.proto
Comment (Tristan1900) [reply proto/import_sstpb.proto]: good catch!

File: proto/encryptionpb.proto
Comment (Tristan1900) [reply proto/encryptionpb.proto:156]: right, it's a placeholder as this proto is going to be serialized and stored as part of the metadata in external storage. The actual plaintext key is going to be passed back by user during restore so not get exposed if external storage is breached.

**Comment by Tristan1900** [comment]
@overvenus thanks for the review! Sorry for the late reply, was doing e2e testing and added one more proto message. Please take an another look if you have time!

**Review by overvenus — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**
